### PR TITLE
Assignment 3.1 all the py file and wheel file for installation

### PR DESCRIPTION
**Pull request by Nikhil-Govindam**
**Assignment 3.1**

All the py file (ingest_data.py, train.py, score.py ) for Training the model
Readme Updated file for model setup 
environment file for environment setup
wheel file 
unit test 
functional test

### CODE REVIEW

**Review by Nikhil-Govindam — COMMENTED** [review]
Assignment 2.1

File: .gitignore
Comment (Nikhil-Govindam) [inline .gitignore:1]: Assignment 2.1

File: ingest_data.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# args = parser.parse_args()
+# train_path = "mle-training"
+
+# Dataset Downloading
+DOWNLOAD_ROOT = "https://raw.githubusercontent.com/ageron/handson-ml/master/"
+HOUSING_PATH = os.path.join(os.getcwd(), "housing")
+HOUSING_URL = DOWNLOAD_ROOT + "datasets/housing/housing.tgz"
+
+
+def fetch_housing_data(housing_url=HOUSING_URL, housing_path=HOUSING_PATH):
```
Comment (vinayksoni) [inline ingest_data.py:38]: datatype

File: ingest_data.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    # summary
```
Comment (vinayksoni) [inline ingest_data.py:39]: doc strings

File: test_package.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+def test_pkg_installation():
```
Comment (vinayksoni) [inline test_package.py:1]: have a separate folder for all the test files and add all checks...

**Review by vinayksoni — CHANGES_REQUESTED** [review]

File: test_file/ingest_data.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    return pd.read_csv(csv_path)
+
+
+housing = load_housing_data()
+
+
+train_set, test_set = train_test_split(housing, test_size=0.2, random_state=42)
+
+
+def file_save():
```
Comment (vinayksoni) [inline test_file/ingest_data.py:63]: py test should not deal with any paths.... once you package this and zip it .... it goes across different systems and different platforms.... and their installation of your packages will fail.

you can copy and create data set (can be a subset) here and then train your model and then get the results and verify the pipeline

File: test_file/score.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+# parser.add_argument("dataset", type=str, help="Dataset")
+# args = parser.parse_args()
+
+
+# model = os.path.join(os.getcwd(), args.final_model)
+# data_path = os.path.join(os.getcwd(), args.dataset)
+# housing = pd.read_csv(data_path)
+final_model = pickle.load(open("/home/nikhil_123/mle-training/model.sav", "rb"))
+
+housing = pd.read_csv("/home/nikhil_123/mle-training/housing.csv")
```
Comment (vinayksoni) [inline test_file/score.py:28]: refer ingestion component

**Review by vinayksoni — CHANGES_REQUESTED** [review]

File: test_file/score.py (hunk 5 above)
Comment (vinayksoni) [reply test_file/score.py:28]: don't use hard coded paths

File: test_file/train.py [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+# Argument Pass for Dataset and Model Save
+# parser = argparse.ArgumentParser(description="Data set path and Folder to save model")
+# parser.add_argument("--housing", type=str, help="Dataset path for model trining")
+# parser.add_argument("--model", type=str, help="path to save model")
+# args = parser.parse_args()
+
+
+# data_path = os.path.join(os.getcwd(), args.housing)
+# housing = pd.read_csv(data_path)
+housing = pd.read_csv("/home/nikhil_123/mle-training/housing.csv")
```
[the remark this answers is not in the document]
Comment (vinayksoni) [reply test_file/train.py:32]: dont use hard coded path

**Review by vinayksoni — CHANGES_REQUESTED** [review]

**Pull request merged**